src/service/resources-impl.js

```diff
@@ -1493,6 +1518,8 @@ export class Resources {
       });
       this.requestsChangeSize_ = this.requestsChangeSize_.filter(
           request => request.resource != resource);
+      this.pendingBuildResources_ = this.pendingBuildResources_.filter(
```

Comment:
Currently you are storing elements in the `pendingBuildResources_`, but here you check for resources. You have to decide which one you store.